assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -178,18 +179,32 @@ def read_data(row):
         return data_url
         # TODO: Move this to tracking stream if both data and dataref are NULL
 
-    # Output MLTable
+    def row_to_pdf(row) -> pd.DataFrame:
+        return pd.read_json(read_data(row))
+
     data_columns = _get_data_columns(df)
-    first_data_row = df.select(data_columns).rdd.map(lambda x: x).first()
+    data_rows = df.select(data_columns).rdd.take(SCHEMA_INFER_ROW_COUNT)  # TODO: make it an argument user can define
```

Comment:
Can we do a sample 10 rows?

Reply:
the sample() only take fraction, no row_count, :(